packages/language-server/src/plugins/typescript/service.ts

```diff
@@ -636,6 +704,26 @@ async function createLanguageService(
 
         docContext.onProjectReloaded?.();
     }
+
+    function updateIfDirty() {
+        if (!dirty) {
+            return;
+        }
+
+        languageService.getProgram();
+        svelteModuleLoader.clearPendingInvalidations();
```

Comment:
This wrapper layer is needed because we need to clear the invalidated files so they won't affect every program update check afterwards. And the hoops you have to jump through are probably why `hasInvalidatedResolutions` is a TypeScript internal API 😅.